packages/flutter_markdown/lib/src/markdown.dart

```diff
@@ -0,0 +1,472 @@
+  MarkdownStyle copyWith({
+    TextStyle a,
+    TextStyle p,
+    TextStyle code,
+    TextStyle h1,
+    TextStyle h2,
+    TextStyle h3,
+    TextStyle h4,
+    TextStyle h5,
+    TextStyle h6,
+    TextStyle em,
+    TextStyle strong,
+    TextStyle blockquote,
+    double blockSpacing,
+    double listIndent,
+    double blockquotePadding,
+    BoxDecoration blockquoteDecoration,
+    double codeblockPadding,
+    BoxDecoration codeblockDecoration
+  }) {
+    return new MarkdownStyle(
+      a: a != null ? a : this.a,
+      p: p != null ? p : this.p,
+      code: code != null ? code : this.code,
+      h1: h1 != null ? h1 : this.h1,
+      h2: h2 != null ? h2 : this.h2,
+      h3: h3 != null ? h3 : this.h3,
+      h4: h4 != null ? h4 : this.h4,
+      h5: h5 != null ? h5 : this.h5,
+      h6: h6 != null ? h6 : this.h6,
+      em: em != null ? em : this.em,
+      strong: strong != null ? strong : this.strong,
+      blockquote: blockquote != null ? blockquote : this.blockquote,
+      blockSpacing: blockSpacing != null ? blockSpacing : this.blockSpacing,
+      listIndent: listIndent != null ? listIndent : this.listIndent,
+      blockquotePadding: blockquotePadding != null ? blockquotePadding : this.blockquotePadding,
+      blockquoteDecoration: blockquoteDecoration != null ? blockquoteDecoration : this.blockquoteDecoration,
+      codeblockPadding: codeblockPadding != null ? codeblockPadding : this.codeblockPadding,
+      codeblockDecoration: codeblockDecoration != null ? codeblockDecoration : this.codeblockDecoration
+    );
+  }
+
+  final TextStyle a;
+  final TextStyle p;
+  final TextStyle code;
+  final TextStyle h1;
+  final TextStyle h2;
+  final TextStyle h3;
+  final TextStyle h4;
+  final TextStyle h5;
+  final TextStyle h6;
+  final TextStyle em;
+  final TextStyle strong;
+  final TextStyle blockquote;
+  final double blockSpacing;
+  final double listIndent;
+  final double blockquotePadding;
+  final BoxDecoration blockquoteDecoration;
+  final double codeblockPadding;
+  final BoxDecoration codeblockDecoration;
+
+  Map<String, TextStyle> _styles;
+
+  void _init() {
+    _styles = {
+      'a': a,
+      'p': p,
+      'li': p,
+      'code': code,
+      'pre': p,
+      'h1': h1,
+      'h2': h2,
+      'h3': h3,
+      'h4': h4,
+      'h5': h5,
+      'h6': h6,
+      'em': em,
+      'strong': strong,
+      'blockquote': blockquote
+    };
+  }
+}
+
+class Markdown extends StatefulComponent {
+  Markdown({
+    this.data,
+    this.markdownStyle,
+    this.syntaxHighlighter
+  });
+
+  final String data;
+  final MarkdownStyle markdownStyle;
+  final SyntaxHighlighter syntaxHighlighter;
+
+  State<Markdown> createState() => new _MarkdownState();
+}
+
+class _MarkdownState extends State<Markdown> {
+
+  void initState() {
+    super.initState();
+
+    MarkdownStyle markdownStyle = config.markdownStyle;
+    if (markdownStyle == null)
+      markdownStyle = new MarkdownStyle.defaultFromTheme(Theme.of(context));
```

Comment:
I can see the value of integrating with material.dart, but you should be able to use Markdown without pulling in material.dart as a dependency.  Maybe the best way to do this is to split the markdown library into two files: one that has the core markdown engine and has no dependency on material.dart and other that has a dependency on material.dart and provides you default theming.

When we've done this in other places in Flutter, we've used the following convention: 

 * RawMarkdown.  This widget requires an explicit MarkdownStyle.
 * Markdown. This widget can be given an explicit style but if you don't provide an explicit style, it will create one automatically based on the current material design Theme.